locator/tablets.cc

```diff
@@ -700,6 +722,71 @@ class tablet_effective_replication_map : public effective_replication_map {
             , _sharder(*_tmptr, table)
     { }
 
+    // Populate the `_tablet_to_dc_replication_factor_map` with replication factors for each data center.
+    // For each tablet index in the tablet map, it retrieves the current replica set using `get_endpoints_for_reading` 
+    // and counts replicas per data center.  Each data center's replication factor (RF) is saved under 
+    // its index according to `_datacenter_map`.
+    seastar::future<> build_dc_replication_factor_map() {
+        const tablet_map& tablets = get_tablet_map();
+        const topology& topo = get_topology();
+        fill_datacenter_map();
+        _tablet_to_dc_replication_factor_map.reserve(tablets.tablet_count());
+        for (size_t tablet_index = 0; tablet_index < tablets.tablet_count(); ++tablet_index) {
+            const tablet_replica_set& replicas = get_endpoints_for_reading(tablet_id{tablet_index});
+            _tablet_to_dc_replication_factor_map.emplace_back();
+            auto& token_rf_list = _tablet_to_dc_replication_factor_map.back();
+            token_rf_list.resize(_datacenter_map.size(), 0);
+            for (const auto& node : replicas) {
+                const locator::node* node_ptr = topo.find_node(node.host);
+                if (node_ptr == nullptr) {
+                    continue;
+                }
+                const sstring& datacenter = topo.get_datacenter(node.host);
+                const auto it = _datacenter_map.find(datacenter);
+                if (it == _datacenter_map.end()) {
+                    tablet_logger.debug("Could not find datacenter: {} for node: {} ", datacenter, node);
```

Comment:
That's a rather serious error because it will result in an inconsistency between replication factor and read replica set. So the problem the patch tries to fix. And this one if happens, may be permanent. Probably deserves on_internal_error().

Reply:
But sometimes this is the case, due to a short time of inconsistency between the Tablets and the schema.

Reply:
But this code is performing lookup in the `locator::topology` data structure. This data structure is completely independent from any schema definitions. It's based purely on topology state. It should never be inconsistent with tablets.

Am I missing something?

Reply:
I did not investigate much, but I was getting these errors in the tests consistently, when ERM was created before the node was initialized.